includes/utils/class-amp-dom-utils.php

```diff
@@ -64,8 +64,21 @@ public static function get_dom( $document ) {
 
 		$dom = new DOMDocument();
 
-		// @todo In the future consider an AMP_DOMDocument subclass that does this automatically. See <https://github.com/Automattic/amp-wp/pull/895/files#r163825513>.
-		$document = self::convert_amp_bind_attributes( $document );
+		/**
+		 * Convert Bind Attributes.
+		 *
+		 * Use this filter to skip converting amp-bind style attributes. On large documents that
+		 * are known not to contain amp-bind style attributes, it's more performant to skip the
+		 * conversion.  Use of this filter is only recommended if documents are very large and
+		 * at risk of exceeding regex engine limits.
+		 *
+		 * @param bool   $convert  True if bind attributes should be converted, false otherwise. Default true.
+		 * @param string $document Valid HTML document possibly containing bind attributes to be converted.
+		 */
+		if ( apply_filters( 'amp_convert_bind_attributes', true, $document ) ) {
```

Comment:
So you see this being necessary? Is it required in your use case? How big are the HTML documents you're working with?

Reply:
The HTML document which led to this error was the only one I've identified, so my sample size is small lol.

In my particular case, without this filter, the document will be passed through the conversion method and `DOMDocument` will be able to load it.  This filter is not strictly necessary, I just thought it was a convenient way for a developer to bypass an operation that they know will not serve a purpose.  That said, I don't want to pollute the codebase with obscure filters that might be misused.  I'll leave it up to you to decide whether it should be included.  I'm happy to remove it.

Reply:
I re-thought this, this filter only confuses the issue.  I'll remove, I don't see a strong enough use case to include it.